### ISSUE TITLE
Error during setup of component ourgroceries (regex change needed, PR submitted to py-our-groceries)

### ISSUE THREAD

**Issue by neilmartin83** [ljmerza/ha-our-groceries#36]
This is just to make folks aware that there's been a new update to the Our Groceries site today which breaks the py-our-groceries python module (that this component uses).

The Home Assistant component won't start and an error will be shown in the logs:

```
Logger: homeassistant.setup
Source: custom_components/ourgroceries/__init__.py:66
Integration: ourgroceries
First occurred: 10:27:47 (1 occurrences)
Last logged: 10:27:47

Error during setup of component ourgroceries
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 256, in _async_setup_component
    result = await task
  File "/config/custom_components/ourgroceries/__init__.py", line 66, in async_setup
    await og.login()
  File "/usr/local/lib/python3.10/site-packages/ourgroceries/__init__.py", line 109, in login
    await self._get_master_list_id()
  File "/usr/local/lib/python3.10/site-packages/ourgroceries/__init__.py", line 151, in _get_master_list_id
    self._master_list_id = re.findall(REGEX_MASTER_LIST_ID, responseText)[0]
IndexError: list index out of range
```

I've submitted the Pull Request linked below to fix it but raising an issue here so folks are aware.

https://github.com/ljmerza/py-our-groceries/pull/20

**Comment by bschatzow** [comment]
Thanks. I was just going to open this.

**Comment by ljmerza** [comment]
fix in v1.5.3

**Comment by lxqikrana** [comment]
I see the same error occurring again:
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 288, in _async_setup_component
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/ourgroceries/__init__.py", line 66, in async_setup
    await og.login()
  File "/usr/local/lib/python3.11/site-packages/ourgroceries/__init__.py", line 109, in login
    await self._get_master_list_id()
  File "/usr/local/lib/python3.11/site-packages/ourgroceries/__init__.py", line 151, in _get_master_list_id
    self._master_list_id = re.findall(REGEX_MASTER_LIST_ID, responseText)[0]
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range

Looks like the regex has changed again? I tried reinstall the ourgroceries integration from HACS and I'm running v1.5.3, so this appears to be a new problem.

**Comment by pbear7656977** [comment]
+1 as also experiencing same issue and tried same actions as lxqikrana above.

**Comment by neilmartin83** [comment]
I've submitted a new PR https://github.com/neilmartin83/py-our-groceries/commit/7f41f3e3f932e39f985422264d8b6d112ddc8059 which hopefully will sort this but the owner will need to accept and merge it then release a new version of the integration before we can use it.

**Comment by bschatzow** [comment]
@neilmartin83 I tried your changes and they did not work for me.

**Comment by lxqikrana** [comment]
@bschatzow @neilmartin83 I tried a similar edit on a standalone py-our-groceries checkout and while it passes the initial REGEX_MASTER_LIST_ID error that HA shows, it dies in a different location. I cannot figure out how to modify my local version of py-our-groceries on HAOS - ssh only gives me access to the ourgroceries HA component. 

Side suggestion - give that ourgroceries keeps changing the format regularly, is there a way to make the regexes that break user configurable? So we can contribute community fixes if needed?
